pallets/base-nft/src/lib.rs

```diff
@@ -231,33 +231,39 @@ impl<T: Config> Pallet<T> {
 
 	/// Transfer NFT(non fungible token) from `from` account to `to` account
 	pub fn transfer(
-		from: &T::AccountId,
-		to: &T::AccountId,
+		from: Vec<&T::AccountId>,
+		to: Vec<&T::AccountId>,
 		token: (T::ClassId, T::TokenId),
+		fraction_amount: u8,
 	) -> DispatchResult {
 		Tokens::<T>::try_mutate(token.0, token.1, |token_info| -> DispatchResult {
 			let mut info = token_info.as_mut().ok_or(Error::<T>::TokenNotFound)?;
-			ensure!(info.owner == *from, Error::<T>::NoPermission);
+			ensure!(
+				info.owners.iter().any(|v| from.iter().any(|w| w == &v)),
+				Error::<T>::NoPermission
+			);
+
 			if from == to {
 				// no change needed
 				return Ok(());
 			}
 
-			info.owner = to.clone();
+			// TODO: assign partial ownership according to fraction_amount
```

Comment:
It goes without saying there is more implementation to do here in this function as well as the others.